setup.py

```diff
@@ -32,12 +39,26 @@
     ],
 }
 
-extras_require['dev'] = (
-    extras_require['dev'] +
-    extras_require['test'] +
-    extras_require['lint'] +
-    extras_require['doc']
-)
+
+def include_extras(base, includes):
```

Comment:
Once we start doing sequential extra definitions, we can pass in the dependencies explicitly here (rather than force them to be named ahead of time in the extras).

I'm thinking an API like `include_extras(base_list, *supplemental_lists)`:
```py
hypothesis_requirement = "hypothesis>=3.6.1"

extras_require['tools'] = [hypothesis_requirement]
extras_require['test'] = [
    "pytest...",
    ...,
    hypothesis_requirement,
]

extras_require['dev'] = include_extras(
  [
    "bumpversion...",
    ...,
  ],
  extras_require['test'],
  extras_require['lint'],
  extras_require['doc'],
)
```

Reply:
Yeah, I like that better.  I think it was the mutable, closure-centered nature of how I was doing it that bothered me.